### PR TITLE
Change test templating syntax to be valid C

### PR DESCRIPTION
**Pull request by davidhorstmann-arm**
For the benefit of auto-formatting tools, move from the `$placeholder`
templating syntax to a new syntax of the form `__MBEDTLS_TEST_TEMPLATE__PLACEHOLDER`.

This change allows the test code template to be almost entirely valid C.

This is mostly to appease Uncrustify, which is confused by the `$` syntax.

For more information on the Python string `Template` class along with descriptions of the special regex groups 'escaped', 'named', 'braced' and 'invalid', see the documentation for `Template` [here](https://docs.python.org/3/library/string.html#string.Template).

Backport: #6536

- [ ] ~**changelog** provided, or~ not required
- [x] **backport** done, or not required
- [ ] ~**tests** provided, or~ not required

### CODE REVIEW

**Comment by davidhorstmann-arm** [comment]
Force-pushed to remove accidental style changes in commit

File: tests/scripts/generate_test_code.py [hunk 1]
```diff
@@ -126,9 +126,9 @@
 This script replaces following fields in the template and generates
 the test source file:
 
-$test_common_helpers        <-- All common code from helpers.function
+/*%test_common_helpers%*/   <-- All common code from helpers.function
```
Comment (gilles-peskine-arm) [inline tests/scripts/generate_test_code.py:129]: Why not use valid C identifier as markers? This should avoid confusing any tool that parses C, and for tools that work after preprocessing, we could potentially `-D` those identifiers.

**Review by gilles-peskine-arm — APPROVED** [review]
Looks mostly good to me, with only non-blocker remarks. Feel free to address them or not. (I think might as well, since the backport will need a review and CI cycle anyway.)

In addition to review, I've verified that the generated `.c` and `.datax` files are the same before and after.

**Review by gilles-peskine-arm — APPROVED** [review]
LGTM

**Review by tom-cosgrove-arm — APPROVED** [review]
LGTM

**Pull request merged**